app/controllers/users_controller.rb

```diff
@@ -0,0 +1,98 @@
+class UsersController < ApplicationController
+  before_action :authorize_supervisor
+  before_action :set_default_table_sort_options, only: %i[index]
+
+  DEFAULT_SORT = 'name'.freeze
+
+  def index
+    pagy, users = pagy_array(sorted_users)
+
+    render :index, locals: { pagy:, users: }
+  end
+
+  def new
+    @form_object = UserForm.new
+  end
+
+  def edit
+    @user = User.find(controller_params['id'])
+    user = @user.attributes.slice('first_name', 'last_name', 'email')
```

Comment:
nitpick: not sure if there's existing reccomendations for this type of thing but having the instance variable and local variable be the same name could get confusing (i.e. people might think the local variable = the instance variable when diagnosing. That said, there isn't really a better name for the variable so not really sure what the best way is to go forward, could call it form_user or something but that also is awkward. Maybe we just leave it as is (still going to leave this comment in case you have any ideas)

Reply:
Don't think there's really much better as you say